src/main/java/org/libra/Testnet.java

```diff
@@ -31,68 +30,69 @@
     public static ChainId CHAIN_ID = new ChainId((byte) 2);
     public static String DD_ADDRESS = "000000000000000000000000000000DD";
 
-    private static final long DEFAULT_TIMEOUT = 10 * 1000;
+    private static final int DEFAULT_TIMEOUT = 10 * 1000;
 
+    /**
+     * Create a LibraClient connects to Testnet.
+     *
+     * @return `LibraClient`
+     */
     public static LibraClient createClient() {
         return new LibraJsonRpcClient(JSON_RPC_URL, CHAIN_ID);
     }
 
+    /**
+     * Mint coins for given authentication key derived account address.
+     *
+     * @param client       a client connects to Testnet
+     * @param amount       amount of coins to mint
+     * @param authKey      authentication key of the account, if account does not exist onchain, a new onchain account will be created.
+     * @param currencyCode currency code of the minted coins
+     */
     public static void mintCoins(LibraClient client, long amount, String authKey, String currencyCode) {
-        long nextAccountSeq = mintCoinsAsync(amount, authKey.toLowerCase(), currencyCode);
-        JsonRpc.Transaction txn;
+        Retry<Integer> retry = new Retry<>(10, 500, Exception.class);
         try {
-            txn = waitForTransaction(DD_ADDRESS, nextAccountSeq - 1, false, DEFAULT_TIMEOUT, client);
+            retry.execute(() -> {
+                List<SignedTransaction> txns = mintCoinsAsync(amount, authKey.toLowerCase(), currencyCode);
+                for (SignedTransaction txn : txns) {
+                    client.waitForTransaction(txn, DEFAULT_TIMEOUT);
+                }
+                return 0;
+            });
         } catch (Exception e) {
-            throw new RuntimeException(e);
-        }
-
-        if (txn == null) {
-            throw new RuntimeException("mint coins transaction does not exist / failed, sequence: " + nextAccountSeq);
-        }
-        if (!TransactionUtils.isExecuted(txn)) {
-            throw new RuntimeException("mint coins transaction failed: " + txn.toString());
+            throw new RuntimeException("Mint coins failed", e);
```

Comment:
I think you want an named exception , right? similar to ClientException , otherwise I don't think user of the SDK can catch this effectively

Reply:
Except the Mint function, this should only be used for testing purpose, so most likely we don't need named exception and handle this error, when it happens, test failed.